backend/typescript/services/implementations/campService.ts

```diff
@@ -440,37 +442,72 @@ class CampService implements ICampService {
-          $set: {
-            name: camp.name,
-            active: camp.active,
-            ageLower: camp.ageLower,
-            ageUpper: camp.ageUpper,
-            campCoordinators: camp.campCoordinators,
-            campCounsellors: camp.campCounsellors,
-            description: camp.description,
-            earlyDropoff: camp.earlyDropoff,
-            latePickup: camp.latePickup,
-            dropoffFee: camp.dropoffFee,
-            pickupFee: camp.pickupFee,
-            location: camp.location,
-            startTime: camp.startTime,
-            endTime: camp.endTime,
-            volunteers: camp.volunteers,
-            fee: camp.fee,
-            fileName: camp.filePath || oldCamp.fileName ? fileName : null,
+      if (
+        camp.earlyDropoff &&
+        camp.latePickup &&
+        camp.pickupFee &&
+        camp.dropoffFee
+      ) {
+        newCamp = await MgCamp.findByIdAndUpdate(
+          campId,
+          {
+            $set: {
+              name: camp.name,
+              active: camp.active,
+              ageLower: camp.ageLower,
+              ageUpper: camp.ageUpper,
+              campCoordinators: camp.campCoordinators,
+              campCounsellors: camp.campCounsellors,
+              description: camp.description,
+              earlyDropoff: camp.earlyDropoff,
+              latePickup: camp.latePickup,
+              dropoffFee: camp.dropoffFee,
+              pickupFee: camp.pickupFee,
+              location: camp.location,
+              startTime: camp.startTime,
+              endTime: camp.endTime,
+              volunteers: camp.volunteers,
+              fee: camp.fee,
+              fileName: camp.filePath || oldCamp.fileName ? fileName : null,
+            },
           },
-        },
-        {
-          new: true,
-          session,
-        },
-      ).populate({
-        path: "campSessions",
-        model: MgCampSession,
-      });
+          {
+            new: true,
+            session,
+          },
+        ).populate({
+          path: "campSessions",
+          model: MgCampSession,
+        });
+      } else {
+        newCamp = await MgCamp.findByIdAndUpdate(
+          campId,
+          {
+            $set: {
+              name: camp.name,
+              active: camp.active,
+              ageLower: camp.ageLower,
+              ageUpper: camp.ageUpper,
+              campCoordinators: camp.campCoordinators,
+              campCounsellors: camp.campCounsellors,
+              description: camp.description,
+              location: camp.location,
+              startTime: camp.startTime,
+              endTime: camp.endTime,
+              volunteers: camp.volunteers,
+              fee: camp.fee,
+              fileName: camp.filePath || oldCamp.fileName ? fileName : null,
+            },
+          },
+          {
+            new: true,
+            session,
+          },
+        ).populate({
+          path: "campSessions",
+          model: MgCampSession,
```

Comment:
how does this handle updates to formQuestions?

Reply:
We do this for updating formQuestions:

```
// Update the FormQuestions by deleting all the current questions and creating new ones
    const currFormQuestions = (newCamp.formQuestions as FormQuestion[]).map(
      (fq) => fq.toString(),
    );
    await this.deleteFormQuestionsByIds(currFormQuestions, session);
    newFormQuestions = await this.addFormQuestionsToCamp(
      newCamp.id,
      camp.formQuestions,
      session,
    );
```

Although I think we can remove this populate functionality since these camp sessions are no longer the updated camp sessions